src/chaos.py

```diff
@@ -73,9 +74,25 @@ def get_targets(autoscaling, default_probability):
     return targets
 
 
-def terminate_targets(ec2, targets):
+def send_notification(sns, instance_id, asg_name):
+    topic = os.environ.get("termination_topic_arn", "").strip()
+    if topic == '':
+        return
+    notification = {
+        "event_name": "chaos_lambda.terminating",
+        "instance_id": instance_id,
+        "asg_name": asg_name,
+    }
+    sns.publish(
+        TopicArn=topic,
+        Message=json.dumps(notification)
+    )
+
+
+def terminate_targets(ec2, sns, targets):
     for asg_name, instance_id in targets:
         log("targeting", instance_id, "in", asg_name)
+        send_notification(sns, instance_id, asg_name)
```

Comment:
Can we `try/except` sending this notification. This way we can log any failures to send a notification, but not interrupt the actual instance termination. Just because chaos-lambda is kind of an "invisible" component in people's AWS accounts and if it stops terminating instances due to errors (because of SNS permissions etc) it's easy for people to not notice it. If we handle any SNS errors at least chaos-lambda would continue to kill instances ok.